scripts/kiwix-resources

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+
+'''
+Copyright 2022 Veloman Yunkan <veloman.yunkan@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3 of the License, or any
+later version.
+
+This program is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+02110-1301, USA.
+'''
+
+import argparse
+import hashlib
+import os.path
+import re
+
+def read_resource_file(resource_file_path):
+    with open(resource_file_path, 'r') as f:
+        return [line.strip() for line in f]
+
+def list_resources(resource_file_path):
+    for resource_path in read_resource_file(resource_file_path):
+        print(resource_path)
+
+def get_resource_revision(base_dir, resource_path):
+    with open(os.path.join(base_dir, resource_path), 'rb') as f:
+        return hashlib.sha1(f.read()).hexdigest()[:8]
+
+resource_revisions = {}
+
+def fill_resource_revisions(resource_file_path):
+    base_dir = os.path.dirname(os.path.realpath(resource_file_path))
+    for resource in read_resource_file(resource_file_path):
+        resource_revisions[resource] = get_resource_revision(base_dir, resource)
+
+src_or_href_pattern=r'(src|href)="\{\{root\}\}/(skin/[^"]+)"'
+def add_resource_revision(resource_matchobj):
+    attr = resource_matchobj.group(1)
+    resource = resource_matchobj.group(2)
+    if resource in resource_revisions:
+        uniqueid = '?' + resource_revisions[resource]
```

Comment:
If the matched link already contains a query string, this will create a invalid link.
I don't think it is important for now as the resource links are pointing to "file" but I prefer to mention it now.

Reply:
The hack with `KIWIXCACHEID` solves this potential issue as well